### ISSUE TITLE
Drop support for CONF_FILE

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#13715]
Right now you can configure Elasticsearch to read its configuration from either a `CONF_FILE` or a `CONF_DIR` but the `CONF_FILE` directive isn't consistently implemented because its default (`$CONF_DIR/elasticsearch.yml`) is finicky to get right.

So lets remove it in the spirit of simplification. See discussion about it in #13687 for more.

**Comment by brwe** [comment]
Are we only talking about CONF_FILE (which is only for services) or also about `-Des.conf=/path/to/conf.yml`?

**Comment by nik9000** [comment]
Both seem redundant. `CONF_FILE` is the services parlance for `-Des.conf` I believe.

**Comment by kimchy** [comment]
I am +1 on removing the conf file in ES itself, and just rely on custom conf dir, but then if we can have a good logging message when someone provides it to explain that this is not used, or maybe even exit (in Bootstrap?) if it is specified.

**Comment by brwe** [comment]
Turns out it is not so easy to exit in Bootstrap when people run elasticsearch service. For services   convert a variable `CONF_FILE` to `-Des.default.conf` and [pass this to elasticsearch] (https://github.com/elastic/elasticsearch/blob/master/distribution/deb/src/main/packaging/init.d/elasticsearch#L99). So we have to check in the init scripts too if `CONF_FILE` was set. Otherwise someone who configured that before will install elasticsearch, start as usual but the new installation will silently pick up the default conf.
For this reason I added a check in all init scripts in #13772  but that is very ugly as @spinscale pointed out.

We thought the following would be a trade-of:
- push the ugly changes in #13772 with all the additional checks to 2.0 and 2.x so that there are no bad surprises for people who use the `CONF_FILE` option and upgrade
- remove `CONF_FILE` and the additional checks completely on master
